test/system-tests/request_handler.cpp

```diff
@@ -240,9 +221,34 @@ void RequestHandler::on_inject_headers(const httplib::Request& req,
   res.set_content(response_json.dump(), "application/json");
 }
 
+void RequestHandler::on_extract_headers(const httplib::Request& req,
+                                        httplib::Response& res) {
+  const auto request_json = nlohmann::json::parse(req.body);
+  auto http_headers = utils::get_if_exists<nlohmann::json::array_t>(
+      request_json, "http_headers");
+  if (!http_headers) {
+    VALIDATION_ERROR(res, "on_extract_headers: missing `http_headers` field.");
+  }
+
+  datadog::tracing::SpanConfig span_cfg;
+  auto span =
+      tracer_.extract_span(utils::HeaderReader(*http_headers), span_cfg);
```

Comment:
This operation creates a child span from distributed tracing headers. This is not what we need. 

We need a way to extract distributed tracing headers and store it in a "SpanContext" or a local representation of the remote span (ex: MergedContext or a "RemoteSpan"). 

To achieve this functionality we need to modify the cpp tracer (ie add an extract_headers(...) function to the public API). The cpp tracer is the only tracer that does not provide this API.